charactersheet/charactersheet/services/common/sync/card_service.js

```diff
@@ -122,18 +122,16 @@ function _pCardService(configuration) {
         }
         var pCardInParty = false;
         var newPCard = pCard.fromEntries(inputPCard);
-        var publisherJid = newPCard.get('publisherJid')[0].split('@')[0];
-        var players = Object.keys(chat.rooms[chat.currentPartyNode].roster);
-        if (players.length > 0) {
-            players.forEach(function(player, idx, _) {
-                if (player === publisherJid) {
-                    pCardInParty = true;
-                }
-            });
-        }
+        var publisherJid = newPCard.get('publisherJid')[0];
+        var players = chat.getOccupantsInRoom(chat.currentPartyNode);
+        players.forEach(function(player, idx, _) {
```

Comment:
I notice you do this pattern a lot.

```
var someBool = false
for item in list:
  if item == 4:
    someBool == true
```

You can accomplish the same thing in a bit more concise way like this:

```js 
// array.some returns true if any function returns true.
var pCardInParty = players.some(function(player, idx, _) {
  return player == publisherJid;
});
```

This way you don't have these imperative loops everywhere.